Code/Framework/AzFramework/AzFramework/Spawnable/SpawnableAssetRef.cpp

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzCore/RTTI/BehaviorContext.h>
+#include <AzCore/Serialization/EditContext.h>
+#include <AzCore/Serialization/SerializeContext.h>
+#include <AzFramework/Spawnable/SpawnableAssetHandler.h>
+#include <AzFramework/Spawnable/SpawnableAssetRef.h>
+#include <AzFramework/StringFunc/StringFunc.h>
+
+namespace AzFramework
+{
+    void SpawnableAssetRef::Reflect(AZ::ReflectContext* context)
+    {
+        if (auto serializeContext = azrtti_cast<AZ::SerializeContext*>(context))
+        {
+            serializeContext
+                ->Class<SpawnableAssetRef>()
+                ->Field("Asset", &SpawnableAssetRef::m_asset);
+
+            serializeContext->RegisterGenericType<AZStd::vector<SpawnableAssetRef>>();
+            serializeContext->RegisterGenericType<AZStd::unordered_map<AZStd::string, SpawnableAssetRef>>();
+            serializeContext->RegisterGenericType<AZStd::unordered_map<double, SpawnableAssetRef>>(); // required to support Map<Number, SpawnableAssetRef> in Script Canvas
+
+            if (AZ::EditContext* editContext = serializeContext->GetEditContext())
+            {
+                editContext
+                    ->Class<SpawnableAssetRef>("SpawnableAssetRef", "A wrapper around spawnable asset to be used as a variable in Script Canvas.")
+                    // m_asset
+                    ->DataElement(AZ::Edit::UIHandlers::Default, &SpawnableAssetRef::m_asset, "asset", "")
+                    ->Attribute(AZ::Edit::Attributes::ShowProductAssetFileName, false)
+                    ->Attribute(AZ::Edit::Attributes::HideProductFilesInAssetPicker, true)
+                    ->Attribute(AZ::Edit::Attributes::AssetPickerTitle, "Spawnable Asset")
+                    ->Attribute(AZ::Edit::Attributes::ChangeNotify, &SpawnableAssetRef::OnSpawnAssetChanged);
+            }
+        }
+
+        if (auto behaviorContext = azrtti_cast<AZ::BehaviorContext*>(context))
+        {
+            behaviorContext
+                ->Class<SpawnableAssetRef>("SpawnableAssetRef")
+                ->Constructor()
+                ->Attribute(AZ::Script::Attributes::Scope, AZ::Script::Attributes::ScopeFlags::Common)
+                ->Attribute(AZ::Script::Attributes::Category, "Prefab/Spawning")
+                ->Attribute(AZ::Script::Attributes::Module, "prefabs")
+                ->Property("asset", BehaviorValueProperty(&SpawnableAssetRef::m_asset));
+        }
+    }
+
+    SpawnableAssetRef::SpawnableAssetRef()
+    {
+        OnSpawnAssetChanged();
```

Comment:
m_asset would be empty on a default constructor. Why call 'OnSpawnAssetChanged' then? Maybe we can delete the default constructor altogether and only have constructors that require passing an asset ref?

Reply:
I need empty ctor for reflection, I've deleted OnSpawnAssetChanged